org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/src_annotationCodeAction.java

```diff
@@ -0,0 +1,59 @@
+package org.eclipse.lemminx.extensions.contentmodel.participants.codeactions;
+
+import java.util.List;
+
+
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lemminx.commons.BadLocationException;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.dom.DOMElement;
+import org.eclipse.lemminx.dom.DOMNode;
+import org.eclipse.lemminx.services.extensions.codeaction.ICodeActionParticipant;
+import org.eclipse.lemminx.services.extensions.codeaction.ICodeActionRequest;
+
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lemminx.commons.CodeActionFactory;
+
+public class src_annotationCodeAction implements ICodeActionParticipant {
+
+	@Override
+	public void doCodeAction(ICodeActionRequest request, List<CodeAction> codeActions, CancelChecker cancelChecker) {
+		Diagnostic diagnostic = request.getDiagnostic();
+		DOMDocument document = request.getDocument();
+		Range diagnosticRange = diagnostic.getRange();
+		String codeActionText;
+
+		// Attempt to get tag name
+		try {
+			int startOffset = document.offsetAt(diagnosticRange.getStart()) + 1;
+			DOMNode node = document.findNodeAt(startOffset);
+			DOMElement element = (DOMElement) node;
+			String tagName = element.getTagName();
+			codeActionText = "Replace '" + tagName + "' with ";
+		} catch (BadLocationException e) {
+			codeActionText = "Replace with ";
+		}
+		
+	
+		// Replace with "appinfo"
+		CodeAction replaceAction_appinfo = CodeActionFactory.replace(codeActionText + "'xs:appinfo'", diagnosticRange,
```

Comment:
You replace just the start tag name, but you should replace the end tagname (as you have noticed). To do that I suggest that you build a lits of TextEdit with those 2 replacements by using https://github.com/eclipse/lemminx/blob/b0ab7dc60f78095525376ba165cd5299b9421db3/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/commons/CodeActionFactory.java#L108